### PR TITLE
feat(model): optimally implement a uniform real distribution

### PR DESCRIPTION
**Pull request by anarchuser**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,8 +1,7 @@
 #include "main.h"
 
-int main(int argc, char * argv[]) {
+int main() {
     Controller controller = Controller(WIDTH, HEIGHT, ENERGY);
     controller.init();
     controller.run();
-    return 0;
```
Comment (AtomToast) [inline src/main.cpp]: again, I think this breaks the program

**Review by AtomToast — COMMENTED** [review]
There seems to be a lot of overlap with PR #22

**Comment by anarchuser** [comment]
> There seems to be a lot of overlap with PR #22

There is.
The PRs all partly relate to each other, so in order to always provide a compiling version, I needed to make changes in the same part of code in multiple PRs

File: src/main.cpp (hunk 1 above)
Comment (anarchuser) [reply src/main.cpp]: It specifically does not.

for the `int main() {}` function, one _can_ return something, esp. errors or similar, but there is no particular need for it.
The operating system will automatically see the program having run without issue and assign it a 0 - no need for unnecessary things

Comment (anarchuser) [reply src/main.cpp]: and, tbh, a `return 0` isn't particularly fancy or necessary in this part.
so, since it doesn't break anything I removed it

**Comment by AtomToast** [comment]
This makes it pretty hard to figure out the relevant changes for this PR.
Is there a cleaner way to declare a dependency onto another pullrequest @Xerus2000 ?

**Comment by anarchuser** [comment]
> This makes it pretty hard to figure out the relevant changes for this PR.
> Is there a cleaner way to declare a dependency onto another pullrequest @Xerus2000 ?

Just look at the PRs independently.
Each is an in itself working change, and just because they may undo one thing or another, they still all work independently from each other

**Comment by xeruf** [comment]
> Just look at the PRs independently.
> Each is an in itself working change, and just because they may undo one thing or another, they still all work independently from each other

You are basically asking us to do unnecessary work because we review the same code multiple times - please instead only ask for reviews for fully independent PRs!

File: src/main.cpp (hunk 1 above)
Comment (AtomToast) [reply src/main.cpp]: Makes sense. I didn't know if they were required or not since as far as I know they were just part of any standard c++ program

**Comment by anarchuser** [comment]
yeah, sorry
I'll keep it in mind in the future

**Review by AtomToast — APPROVED** [review]

**Pull request merged**